bin/compile

```diff
@@ -196,25 +199,41 @@ fi
 export YARN_CACHE_FOLDER NPM_CONFIG_CACHE
 
 install_bins() {
-  local node_engine npm_engine yarn_engine npm_version node_version
+  local node_engine npm_engine yarn_engine node_version package_manager
 
   node_engine=$(read_json "$BUILD_DIR/package.json" ".engines.node")
   npm_engine=$(read_json "$BUILD_DIR/package.json" ".engines.npm")
   yarn_engine=$(read_json "$BUILD_DIR/package.json" ".engines.yarn")
+  package_manager=$(read_json "$BUILD_DIR/package.json" ".packageManager")
 
   meta_set "node-version-request" "$node_engine"
   meta_set "npm-version-request" "$npm_engine"
   meta_set "yarn-version-request" "$yarn_engine"
+  meta_set "package-manager-request" "$package_manager"
+
+  echo "engines.node (package.json):   ${node_engine:-unspecified}"
+  echo "engines.npm (package.json):    ${npm_engine:-unspecified (use default)}"
 
-  echo "engines.node (package.json):  ${node_engine:-unspecified}"
-  echo "engines.npm (package.json):   ${npm_engine:-unspecified (use default)}"
   if $YARN || [ -n "$yarn_engine" ]; then
-    echo "engines.yarn (package.json):  ${yarn_engine:-unspecified (use default)}"
+    echo "engines.yarn (package.json):   ${yarn_engine:-unspecified (use default)}"
   fi
+
+  if [ -n "$package_manager" ]; then
+    echo "packageManager (package.json): $package_manager"
+  fi
+
   echo ""
 
   warn_node_engine "$node_engine"
 
+  if [ -n "$yarn_engine" ] && [[ "$package_manager" == yarn* ]]; then
+    warn "The Yarn version is specified in both the 'packageManager' and 'engines.yarn' field in package.json. The 'packageManager' field will be preferred and $package_manager will be used."
```

Comment:
`warn` sounds incorrect here -- it usually means the user is doing something incorrect or not recommended. I think it is fine to specify it in both places... is `info` more appropriate here?

Reply:
@joshwlewis I've changed this warning to provide more context and clearer actions.

Reply:
It looks unchanged to me. Did you push your fix?

Reply:
Sorry, was marking these off as I was going through the changes but then got distracted before I could push. Just updated the code a few minutes ago.